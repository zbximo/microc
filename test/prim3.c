void main()
{
    Int i;
    Int j;
    i = 0;
    j = i>1 ? 1 : 0;
    print j;
    j = i<1 ? 1 : 0;
    print j;
}