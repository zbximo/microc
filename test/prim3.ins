LDARGS 0
CALL (0, "L1_main")
STOP
FLabel (0, "L1_main")
INCSP 1
INCSP 1
GETBP
OFFSET 0
ADD
CSTI 0
STI
INCSP -1
GETBP
OFFSET 1
ADD
GETBP
OFFSET 0
ADD
LDI
CSTI 1
IFZERO "L2"
CSTI 1
GOTO "L3"
Label "L2"
CSTI 0
Label "L3"
SWAP
LT
STI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
GETBP
OFFSET 1
ADD
GETBP
OFFSET 0
ADD
LDI
CSTI 1
IFZERO "L4"
CSTI 1
GOTO "L5"
Label "L4"
CSTI 0
Label "L5"
LT
STI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
INCSP -2
RET -1