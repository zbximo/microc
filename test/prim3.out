24 19 0 5 25 15 1 15 1 13 0 0 1 0 0 12 15 -1 13 0 1 1 13 0 0 1 11 0 1 17 35 0 1 16 37 0 0 10 7 12 15 -1 13 0 1 1 11 22 15 -1 13 0 1 1 13 0 0 1 11 0 1 17 67 0 1 16 69 0 0 7 12 15 -1 13 0 1 1 11 22 15 -1 15 -2 21 -1